torchgen/gen.py

```diff
@@ -385,27 +388,31 @@ def add_spl_memory_format_binding(input_bindings: List[Binding]) -> List[Binding
                 output_bindings.append(binding)
         return output_bindings
 
-    disp_sig = DispatcherSignature.from_schema(f.func)
-    cpp_sig = CppSignatureGroup.from_native_function(
+    src_bindings = (
+        list(cpp_sig.arguments())
+        if cpp_sig
+        else DispatcherSignature.from_schema(f.func).arguments()
+    )
+    goals = CppSignatureGroup.from_native_function(
         f, method=False, fallback_binding=False
-    ).signature
-    disp_bindings = disp_sig.arguments()
+    ).signature.arguments()
     # When last argument of CPP signature has SpecialArgName.possibly_redundant_memory_format NCType,
     # get memory_format bindings of dispatcher signature to have the same NCType as well
-    for arg in cpp_sig.arguments():
+    for arg in goals:
         if arg.nctype.name == SpecialArgName.possibly_redundant_memory_format:
-            disp_bindings = add_spl_memory_format_binding(disp_sig.arguments())
+            src_bindings = add_spl_memory_format_binding(src_bindings)
```

Comment:
Can you explain to me why this is needed? Translate knows how to handle redundant memory format correctly, so it is surprising to me that you need to paper over it more here. And even if it didn't, I would expect you to teach translate how to handle it adequately.

Reply:
Yeah good point I'll see what's going on with memory_format. I can follow up in another PR.